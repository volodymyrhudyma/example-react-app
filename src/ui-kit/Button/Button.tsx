import React, { ReactNode } from 'react';
import { Wrapper } from './styled/Button.styled';

interface IButtonProps {
  children: ReactNode;
  onClick?: () => void;
}

const Button = ({ children, ...rest }: IButtonProps) => (
  <Wrapper {...rest}>{children}</Wrapper>
);

export default Button;
