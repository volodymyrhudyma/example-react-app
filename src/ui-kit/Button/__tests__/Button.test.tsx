import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

describe('Button component', () => {
  it('should render correctly', () => {
    const component = shallow(<Button>Button</Button>);
    expect(component).toMatchSnapshot();
  });
});
