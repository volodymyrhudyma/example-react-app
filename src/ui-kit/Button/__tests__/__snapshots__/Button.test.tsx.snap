// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly 1`] = `
<styled.button>
  Button
</styled.button>
`;
